src/libs/deploy.js

```
@@ -0,0 +1,153 @@
+    auth: oauth2Client,
+    params: {
+      packageName: apkName,
+    },
+  });
+
+  try {
+    const token = await jwtClient.authorize();
+    await oauth2Client.setCredentials(token);
+
+    return googleDeployEdit(publisher, editID, signedAPK, trackType);
+  } catch (error) {
+    const message = 'Unable to deploy';
+    logger.error(`${message}, err = ${error.message}`);
+  }
+
+  return Promise.reject();
+};
+
+/**
+ * Google Edit for apk deployment
+ *
+ * @param {*} publisher The google android publisher
+ * @param {String} editID A unique timestamp as a String
+ * @param {*} signedAPK The signed android apk stored in minio
+ * @param {String} trackType The track could be alpha, beta, production, rollout or internal.
+ * @returns The status of the committed the Edit
+ */
+const googleDeployEdit = async (publisher, editID, signedAPK, trackType) => {
+  try {
+    // start a new Google Edit:
+    const newEdit = await publisher.edits.insert({
+      resource: {
+        id: editID,
+        // this edit will be valid for 10 minutes
+        expiryTimeSeconds: 600,
+      },
+    });
+    const newEditID = newEdit.data.id;
+    // Upload the signed apk to the current Edit:
+    const uploadedApk = await publisher.edits.apks.upload({
+      editId: newEditID,
+      media: {
+        mimeType: 'application/vnd.android.package-archive',
+        body: signedAPK,
+      },
+    });
+    // Assign a track for the current Edit:
+    await publisher.edits.tracks.update({
+      editId: newEditID,
+      track: trackType,
+      body: {
+        releases: [{
+          name: 'CICD release',
+          versionCodes: [uploadedApk.data.versionCode],
+          status: 'completed',
+        }],
+      },
+    });
+    // Commit the Edit after all actions done:
+    const commitEdit = await publisher.edits.commit({
+      editId: newEditID,
+    });
+    return Promise.resolve(commitEdit.status);
+  } catch (error) {
+    return Promise.reject(error);
```

I think you want to `throw` an error here rather than return a rejected promise.